light/provider/http/http.go

```diff
@@ -30,39 +32,46 @@ type http struct {
 	maxRetryAttempts int
 }
 
+type Options struct {
+	// -1 means no limit
```

Comment:
should we do this in a separate PR?

Reply:
This PR is probably small enough that we can fit it in. I just have to change `attempt <= p.maxRetryAttempts` to `attempt != p.maxReryAttempts`